### ISSUE TITLE
Model Training Roadmap

### ISSUE THREAD

**Issue by merrymercy** [alpa-projects/alpa#309]
- Features
  - [ ] data loader @merrymercy 
  - [ ] operator coverage (dropout) @merrymercy 
  - [ ] multiple computational graphs (train, eval) @zhuohan123 
  - [x] model checkpoint, initialization @zhuohan123 
  - [x] tied embedding @ZYHowell 
  - [x] pipeline runtime optimization (attention mask, shard args) @ZYHowell 
- Milestones (check numerical correctness)
  - [ ] reproduce e2e resnet training
  - [ ] reproduce e2e bert training

**Comment by merrymercy** [comment]
This is outdated. I will draft a new one.
